pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -639,20 +639,23 @@ private synchronized void resetMetricsServlet() {
     }
 
     private CompletableFuture<Void> addTimeoutHandling(CompletableFuture<Void> future) {
-        ScheduledExecutorService shutdownExecutor = Executors.newSingleThreadScheduledExecutor(
-                new ExecutorProvider.ExtendedThreadFactory(getClass().getSimpleName() + "-shutdown"));
-        FutureUtil.addTimeoutHandling(future,
-                Duration.ofMillis(Math.max(1L, getConfiguration().getBrokerShutdownTimeoutMs())),
-                shutdownExecutor, () -> FutureUtil.createTimeoutException("Timeout in close", getClass(), "close"));
-        future.handle((v, t) -> {
-            if (t != null && getConfiguration().getBrokerShutdownTimeoutMs() > 0) {
-                LOG.info("Shutdown timed out after {} ms", getConfiguration().getBrokerShutdownTimeoutMs());
-                LOG.info(ThreadDumpUtil.buildThreadDiagnosticString());
-            }
-            // shutdown the shutdown executor
-            shutdownExecutor.shutdownNow();
-            return null;
-        });
+        long brokerShutdownTimeoutMs = getConfiguration().getBrokerShutdownTimeoutMs();
```

Comment:
Maybe a guard clause could reduce unnecessary code changes in case there is any minor change in the if block.

```java
if (brokerShutdownTimeoutMs <= 0) {
    return future;
}
```

Reply:
Thanks. Updated. PTAL